### ISSUE TITLE
[BUG]: fh in predict_model does not work as expected 

### ISSUE THREAD

**Issue by mfahadakbar** [pycaret/pycaret#3269]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

if calling predict_model while providing the fh different than what was provided in the setup, we encounter an error. Below is a reproducible example:

```
from pycaret.time_series import TSForecastingExperiment
from pycaret.datasets import get_data
org = TSForecastingExperiment()
data = get_data('airline')
org.setup(data = data, fh = [4,5,6,7], verbose=False)
models = org.compare_models(verbose=False, n_select=3, sort='MAPE')
blend = org.blend_models(estimator_list = models)
org.predict_model(blend,fh=[4,5])
```

```python
ValueError: A different forecasting horizon `fh` has been provided from the one seen in `fit`. If you want to change the forecasting horizon, please re-fit the forecaster. This is because fitting of the `ForecastingPipeline` depends on `fh`. 
```

### Reproducible Example

```python
from pycaret.time_series import TSForecastingExperiment
from pycaret.datasets import get_data
org = TSForecastingExperiment()
data = get_data('airline')
org.setup(data = data, fh = [4,5,6,7], verbose=False)
models = org.compare_models(verbose=False, n_select=3, sort='MAPE')
blend = org.blend_models(estimator_list = models)
org.predict_model(blend,fh=[4,5])
```

### Expected Behavior

It should give a prediction for periods 4 & 5

### Actual Results

```python-traceback
ValueError: A different forecasting horizon `fh` has been provided from the one seen in `fit`. If you want to change the forecasting horizon, please re-fit the forecaster. This is because fitting of the `ForecastingPipeline` depends on `fh`.
```

### Installed Versions

<details>
System:
    python: 3.8.15 (default, Nov 24 2022, 09:04:07)  [Clang 14.0.6 ]
executable: /Users/64323/opt/miniconda3/envs/pycaret4/bin/python
   machine: macOS-10.16-x86_64-i386-64bit

PyCaret required dependencies:
                 pip: 22.3.1
          setuptools: 65.5.0
             pycaret: 3.0.0rc6
             IPython: 8.8.0
          ipywidgets: 8.0.4
                tqdm: 4.64.1
               numpy: 1.23.5
              pandas: 1.5.2
              jinja2: 3.1.2
               scipy: 1.9.3
              joblib: 1.2.0
             sklearn: 1.1.3
                pyod: 1.0.7
            imblearn: 0.10.1
   category_encoders: 2.5.1.post0
            lightgbm: 3.2.1
               numba: 0.56.4
            requests: 2.28.1
          matplotlib: 3.6.2
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.11.0
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.15.0
               tbats: 1.1.2
            pmdarima: 2.0.2
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: Not installed
    pandas_profiling: Not installed
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
             xgboost: Not installed
            catboost: Not installed
              kmodes: Not installed
             mlxtend: Not installed
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: 2.2.0
            hyperopt: Not installed
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
                nltk: Not installed
            pyLDAvis: Not installed
              gensim: Not installed
               spacy: Not installed
           wordcloud: Not installed
            textblob: Not installed
               fugue: Not installed
           streamlit: Not installed
             prophet: Not installed

</details>

**Comment by ngupta23** [comment]
Hi @mfahadakbar , This is the same issue as #2329 and is specific to blending. The individual models work correctly with a different fh. For now, you can do the blending manually by getting the forecasts from the individual models. I will close this in favor of keeping #2329 open.

**Comment by ngupta23** [comment]
@mfahadakbar - Should be fixed when https://github.com/pycaret/pycaret/pull/3534 is merged to master and released (hopefully in 3.0.1)
